core/domain/user_services.py

```diff
@@ -1400,8 +1401,8 @@ def update_email_preferences(
     if email_preferences_model is None:
         email_preferences_model = user_models.UserEmailPreferencesModel(
             id=user_id)
+        email_preferences_model.site_updates = False
 
-    email_preferences_model.site_updates = can_receive_email_updates
```

Comment:
Why have we changed the code here -- shouldn't we be still updating site_updates according to the user's preference? (We're trying to keep the mailchimp and oppia databases in sync, right?)

Reply:
Yeah, I am doing this inside the `add_or_update` function so that they are in-sync, since if it is set here and mailchimp API is called later, which throws an error, then they can go out of sync.

Reply:
Oh, I see! Hm, what do you think about doing the mailchimp call at the beginning instead (i.e. around this line)? Then you can try/except around it and deal with errors correctly if they occur.

Otherwise you run into the opposite problem where some of the user's preferences have been applied but some have not, which could be more confusing?

Reply:
Done, removed the user_models access from the mailchimp function and added a note in the docstring that the caller should ensure synchronization.